burn-ndarray/src/ops/conv.rs

```diff
@@ -84,8 +86,8 @@ fn conv2d_with_kernel<E: NdArrayElement>(
     let [k1, k2] = kernel.shape().dims;
     let [heigth, width] = x.shape().dims;
 
-    let heigth_new = f32::ceil((heigth - k1 + 1) as f32 / stride[0] as f32) as usize;
-    let width_new = f32::ceil((width - k2 + 1) as f32 / stride[1] as f32) as usize;
+    let heigth_new = ceilf((heigth - k1 + 1) as f32 / stride[0] as f32) as usize;
+    let width_new = ceilf((width - k2 + 1) as f32 / stride[1] as f32) as usize;
```

Comment:
I would check here if the bug is not related to the ceiling function. It would be suprising, but hey we don't know.

Reply:
I'll run some tests. I also reverted these functions and the bug persisted.